apis/acmpca/v1alpha1/certificateauthority_types.go

```diff
@@ -0,0 +1,212 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/aws/aws-sdk-go-v2/service/acmpca"
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Tag represents user-provided metadata that can be associated
+type Tag struct {
+
+	// The key name that can be used to look up or retrieve the associated value.
+	Key string `json:"key"`
+
+	// The value associated with this tag.
+	// +optional
+	Value string `json:"value,omitempty"`
+}
+
+// RevocationConfiguration is configuration of the certificate revocation list
+type RevocationConfiguration struct {
+
+	// Boolean value that specifies certificate revocation
+	Enabled *bool `json:"enabled"`
+
+	// Name of the S3 bucket that contains the CRL
+	S3BucketName *string `json:"s3BucketName"`
+
+	// Alias for the CRL distribution point
+	// +optional
+	CustomCname *string `json:"customCname,omitempty"`
+
+	// Number of days until a certificate expires
+	// +optional
+	ExpirationInDays *int64 `json:"expirationInDays,omitempty"`
+}
+
+// CertificateAuthorityConfiguration is
+type CertificateAuthorityConfiguration struct {
+
+	// Type of the public key algorithm
+	// +kubebuilder:validation:Enum=RSA_2048;EC_secp384r1;EC_prime256v1;RSA_4096
+	KeyAlgorithm acmpca.KeyAlgorithm `json:"keyAlgorithm"`
+
+	// Algorithm that private CA uses to sign certificate requests
+	// +kubebuilder:validation:Enum=SHA512WITHECDSA;SHA256WITHECDSA;SHA384WITHECDSA;SHA512WITHRSA;SHA256WITHRSA;SHA384WITHRSA
+	SigningAlgorithm acmpca.SigningAlgorithm `json:"signingAlgorithm"`
+
+	// Subject is information of Certificate Authority
+	Subject Subject `json:"subject"`
+}
+
+// Subject is
+type Subject struct {
+
+	// Organization legal name
+	// +immutable
+	Organization *string `json:"organization"`
+
+	// Organization's subdivision or unit
+	// +immutable
+	OrganizationalUnit *string `json:"organizationalUnit"`
+
+	// Two-digit code that specifies the country
+	// +immutable
+	Country *string `json:"country"`
+
+	// State in which the subject of the certificate is located
+	// +immutable
+	State *string `json:"state"`
+
+	// The locality such as a city or town
+	// +immutable
+	Locality *string `json:"locality"`
+
+	// FQDN associated with the certificate subject
+	// +immutable
+	CommonName *string `json:"commonName"`
```

Comment:
If these are not optional they should not be pointer types.

Reply:
Refactored respected field to remove pointer.